webdriver-spec.html

```diff
@@ -5777,19 +5778,25 @@
  <li><p>Let <var>body</var> and <var>arguments</var>
   be <var>script arguments</var>’ data.
 
- <li><p>Let <var>result</var> be the result of calling <a>execute a function body</a>,
-  with arguments <var>body</var> and <var>arguments</var>.
+ <li><p>Let <var>promise</var> be a new Promise.
 
- <li><p>Wait for <a>execute a function body</a> to return
-  or for <a>session script timeout</a> milliseconds to expire,
-  whichever happens sooner.
+ <li><p>Run the following substeps <a>in parallel</a>:
+   <ol>
+     <li><p>Let <var>result</var> be the result of calling <a>execute a function body</a>,
+      with arguments <var>body</var> and <var>arguments</var>.
 
- <li><p>If the previous step completed due to the <a>session script timeout</a> being reached,
-  return <a>error</a> with <a>error code</a> <a>script timeout</a>.
+     <li><p>If <var>promise</var> is still pending and
+       <a>session script timeout</a> milliseconds is reached,
+      reject <var>promise</var> with <a>error</a> with
+      <a>error code</a> <a>script timeout</a>.
 
- <li><p>If <var>result</var> is an <a>error</a>, return result.
+     <li><p>If <var>result</var> is an <a>error</a>,
+       reject <var>promise</var> with <var>result</var>.
 
- <li><p>Otherwise let <var>value</var> be <var>result</var>’s data.
+     <li><p>Otherwise resolve <var>promise</var> with <var>result</var>.
+   </ol>
+
+ <li><p>Let <var>value</var> be <var>result</var>’s data.
```

Comment:
These last two steps should be written in terms of the resolved promise. Maybe something like:

```html
<li>Upon fulfillment of <var>promise</var> with value <var>v</var>,
    return <a>success</a> with data <var>v</var>.

<li>Upon rejection of <var>promise</var> with reason <var>r</var>,
    if <var>r</var> is an <a>error</a>, return <var>r</var>.
    Otherwise, return <a>error</a> with <a>error code</a> <a>javascript error</a>
    and data <var>r</var>.
```


This follows recommendations in https://www.w3.org/2001/tag/doc/promises-guide#shorthand-reacting something like: